### PR TITLE
Add type definitions

### PR DESCRIPTION
**Pull request by ZaneH**
Type definitions are here! I would consider these types "confidently beta." I tested it against the API manually from the server-side.

I have [this project (branch: test/zabo-types)](https://github.com/ZaneH/zabo-demo/tree/test/zabo-types/zabo-backend-demo/src/controllers) setup with TS and zabo-sdk-js types if you'd like to poke around. Just run `npm run dev` in `./zabo-backend-demo`.

**Note:** To run the demo you must follow [step #1 from the readme](https://github.com/ZaneH/zabo-demo/tree/test/zabo-types#run-demo). You may also need to switch the hardcoded userId & accountId.

## Potential Improvements
- [ ] Type out errors
- [ ] Type out `Teams`
- [ ] Ensure all platform-specific functions are marked
- [ ] `zabo.teams` & `zabo.trading` are undefined on Node (expected?)
- [ ] Export useful types such as Account, User, Currency, Transaction, etc.

### CODE REVIEW

**Review by dreinke — COMMENTED** [review]
Great job! 👏 
Comments in the code...

**Comment by ZaneH** [comment]
Just did a rebase, I'm pretty sure everything went smoothly.
<hr />

File: src/sdk.js [hunk 1]
```diff
@@ -17,157 +17,26 @@
...
-    }).catch(err => {
-      throw err
+/**
+ * SDK main class definition.
+ */
+class SDK extends ZaboSDK {
+  /**
+   * Initialize the Zabo SDK.
+   * @param {Object} params Zabo initialization config.
+   * @returns The Zabo SDK.
```
Comment (marcelodesouza) [inline src/sdk.js:29]: This method initialises the API in SDK and returns nothing.

File: src/api.js [hunk 2]
```diff
@@ -59,6 +61,9 @@ class API {
     this._isWaitingForConnector = false
   }
 
+  /**
+   * Connect to the Zabo API.
+   */
   async connect ({ provider, params } = {}) {
```
Comment (marcelodesouza) [inline src/api.js:67]: Shouldn't we add the @params in the method connect?

**Pull request merged**